### ISSUE TITLE
Text 3.7.2: Files mixed or emptied

### ISSUE THREAD

**Issue by jakobroehrl** [nextcloud/text#4112]
Since NC16 I have some problems with the textfiles. 
I have mainly 2 differnt textfiles that a my users view and some of them edit it. 
I noticed several times that both files are mixed, so the complete contents of both files are stored in both files. 
Or both files are completly empty after a user edited it. 
Till now I cant reproduce it, but I think it's a bug, we work since a long time with this textfiles without this problems.

**Comment by max-nextcloud** [comment]
Hi @jakobroehrl 

We changed the collaboration backend in Nextcloud 26 in order to allow for more people to work on the same file at the same time.

We also noticed some empty files after that and isolated and fixed a problem with the sync mechanism. One fix has been released with 26.0.1. However it seems like the fix did not cover for all cases. We're logging the saving of empty files with a warning from the next release on to be able to better investigate this.

However i have not heard about an issue of mixing the content of two files. That really seems odd. Is there anything special about these files:
* are the Readme.md files?
* are they stored on a external storage?
* are they inside a groupfolder?

Anything that connects these two files?

**Comment by juliushaertl** [comment]
Also can you clarify since when this issue shows? "Since NC16 I have some problems with the textfiles." this sounds like it is an issue for along time already but "we work since a long time with this textfiles without this problems" sounds different.

**Comment by juliushaertl** [comment]
Hi @jakobroehrl With #4100 we found that mobile apps might have been one of the reasons for this happening a potential fix is pushed to #4124. Can you tell if your users used the Android/iOS apps for editing and if that might be related?

**Comment by jakobroehrl** [comment]
Hi Max und Julius,

> are the Readme.md files?

no

> are they stored on a external storage?

no

> are they inside a groupfolder?

no

> Anything that connects these two files?

I think no :)

> Since NC16 I have some problems with the textfiles.

I meant NC26, sorry 

> Can you tell if your users used the Android/iOS apps for editing and if that might be related?

yes, we use a lot the mobile Clients, both Android and iOS.

**Comment by rurseekatze** [comment]
Hello,

since I upgraded my Nextcloud to 26 a few days ago, I also observed the problem of mixed text files described by @jakobroehrl. In my case there were 4 different files mixed together.

The problem ist really annoying as it requires a manual clean up of the files. Luckily it occurred only once here, but that makes it difficult for me to give you any details on how to reproduce the problem. My impression was, that when I opened a text file, the editor window still contained the content of the previously opened file and appended the new text before or below (not sure) the previous text. The file was overwritten with the mixed text when closing the editor.

I am using Text 3.7.2 + Nextcloud Server 26.0.1.
